File: agent/library/src/main/java/org/matilda/messages/MessageHandlerRegistration.java
package org.matilda.messages;

import org.matilda.messages.handlers.MessageHandler;

public class MessageHandlerRegistration {
    private final MessageHandlerRegistry mMessageHandlerRegistry;
    private final int mMessageType;
    private final MessageHandler mMessageHandler;

    public MessageHandlerRegistration(MessageHandlerRegistry messageHandlerRegistry, int messageType,
                                      MessageHandler messageHandler) {
        mMessageHandlerRegistry = messageHandlerRegistry;
        mMessageType = messageType;
        mMessageHandler = messageHandler;
    }

    public void unregister() {
        mMessageHandlerRegistry.unregisterHandler(mMessageType, mMessageHandler);
    }
}

File: agent/library/src/main/java/org/matilda/messages/MessageHandlerRegistration.kt
package org.matilda.messages

import org.matilda.messages.handlers.MessageHandler

class MessageHandlerRegistration(
    private val mMessageHandlerRegistry: MessageHandlerRegistry, private val mMessageType: Int,
    private val mMessageHandler: MessageHandler
) {
    fun unregister() {
        mMessageHandlerRegistry.unregisterHandler(mMessageType, mMessageHandler)
    }
}
